sql/maintenance_utils.sql

```diff
@@ -39,11 +39,24 @@ CREATE OR REPLACE FUNCTION @extschema@.compress_chunk(
     hypercore_use_access_method BOOL = NULL
 ) RETURNS REGCLASS AS '@MODULE_PATHNAME@', 'ts_compress_chunk' LANGUAGE C VOLATILE;
 
+-- Alias for compress_chunk above.
+CREATE OR REPLACE FUNCTION @extschema@.convert_to_columnstore(
```

Comment:
Maybe a bit controversial, but I am wondering if we should take the opportunity to make these new APIs procedures instead. I think that gives more room to do more flexible conversions, e.g., using multiple transactions, to reduce time spent holding locks.

I also think that maintenance/alter-type functions also fit better as procedures given how the rewrite data. It will no longer be possible to do compression on multiple chunks as part of a query, but that might actually be a feature more than a bug.

Reply:
I agree there are benefits in using procedures rather than functions, but we cannot modify the existing functions to be procedures since that would mean a change in the API. We can, however, use procedures for the new functions and/or create procedure versions of the old functions.

Reply:
I only meant doing it for the new API. The existing functions remain as is. But it seems like a lost opportunity to not do procedures in the new API.

Reply:
Functions turned into procedure. I only dealt with "convert_to_..." and not, e.g., policy adding functions. Not sure if we want to change those as well.